appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/interceptors/InterceptorUtil.java

```diff
@@ -37,120 +37,82 @@
-         smallerPrimitiveWrappers.add(Double.class);
-         compatiblePrimitiveWrapper.put(double.class, smallerPrimitiveWrappers);
-
-         smallerPrimitiveWrappers = new HashSet<>();
-         smallerPrimitiveWrappers.add(Byte.class);
-         smallerPrimitiveWrappers.add(Short.class);
-         smallerPrimitiveWrappers.add(Integer.class);
-         smallerPrimitiveWrappers.add(Float.class);
-         compatiblePrimitiveWrapper.put(float.class, smallerPrimitiveWrappers);
-
-         smallerPrimitiveWrappers = new HashSet<>();
-         smallerPrimitiveWrappers.add(Byte.class);
-         smallerPrimitiveWrappers.add(Short.class);
-         smallerPrimitiveWrappers.add(Integer.class);
-         compatiblePrimitiveWrapper.put(int.class, smallerPrimitiveWrappers);
-
-         smallerPrimitiveWrappers = new HashSet<>();
-         smallerPrimitiveWrappers.add(Byte.class);
-         smallerPrimitiveWrappers.add(Short.class);
-         smallerPrimitiveWrappers.add(Integer.class);
-         smallerPrimitiveWrappers.add(Long.class);
-         compatiblePrimitiveWrapper.put(long.class, smallerPrimitiveWrappers);
+    private static final Map<Class<?>, Set<Class<?>>> compatiblePrimitiveWrapper = new ConcurrentHashMap<>();
+
+    static {
+        compatiblePrimitiveWrapper.put(byte.class, asSet(Byte.class));
+        compatiblePrimitiveWrapper.put(boolean.class, asSet(Boolean.class));
+        compatiblePrimitiveWrapper.put(char.class, asSet(Character.class));
+        compatiblePrimitiveWrapper.put(double.class, asSet(Byte.class, Short.class, Integer.class, Float.class, Double.class));
+        compatiblePrimitiveWrapper.put(float.class, asSet(Byte.class, Short.class, Integer.class, Float.class));
+        compatiblePrimitiveWrapper.put(int.class, asSet(Byte.class, Short.class, Integer.class));
+        compatiblePrimitiveWrapper.put(long.class, asSet(Byte.class, Short.class, Integer.class, Long.class));
+        compatiblePrimitiveWrapper.put(short.class, asSet(Byte.class, Short.class));
+    }
 
-         smallerPrimitiveWrappers = new HashSet<>();
-         smallerPrimitiveWrappers.add(Byte.class);
-         smallerPrimitiveWrappers.add(Short.class);
-         compatiblePrimitiveWrapper.put(short.class, smallerPrimitiveWrappers);
-     }
+    private static Set<Class<?>> asSet(Class<?>... classes) {
+        Set<Class<?>> set = ConcurrentHashMap.newKeySet(classes.length);
+        for (Class<?> c : classes) {
+            set.add(c);
+        }
+        return set;
+    }
 
-    public static boolean hasCompatiblePrimitiveWrapper(Class type, Class typeTo) {
-        Set<Class> compatibles = compatiblePrimitiveWrapper.get(type);
-        return compatibles.contains(typeTo);
+    public static boolean hasCompatiblePrimitiveWrapper(Class<?> type, Class<?> typeTo) {
+        Set<Class<?>> compatibles = compatiblePrimitiveWrapper.get(type);
+        return compatibles != null && compatibles.contains(typeTo);
```

Comment:
Another theoretical NPE here in case method is called with a non-primitive `type` value.